src/main/webapp/js/components/csv/csv-config-page.jsx

```diff
@@ -0,0 +1,259 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { FieldArray } from 'react-final-form-arrays';
+import { Form, Field } from 'react-final-form';
+import arrayMutators from 'final-form-arrays';
+import 'react-table/react-table.css';
+import { withRouter } from 'react-router-dom';
+
+import renderFormField from '../../utils/form/form-utils';
+import {
+  TEXT,
+  ENUM,
+  RELATION,
+  DATE,
+  DATE_TIME,
+} from '../../constants/field-types';
+import { getFieldConfigByEntity } from '../../selectors';
+import TextField from '../../utils/form/text-field';
+
+class CsvConfigPage extends Component {
+  onSubmit = (values) => {
+    const csvConfig = values;
+    csvConfig.entity = this.props.entityType;
+    this.props.onSubmit(csvConfig);
+  };
+
+  getOptions = () => _.map(this.props.fieldConfig, v => ({
+    label: v.displayName,
+    value: v.id,
+  }));
+
+  getFields = (fieldValue) => {
+    let hideFormat = true;
+
+    // eslint-disable-next-line max-len
+    const fieldConfig = this.props.fieldConfig[fieldValue.fieldConfigId];
+    if (fieldConfig && (fieldConfig.fieldType === DATE || fieldConfig.fieldType === DATE_TIME)) {
+      hideFormat = false;
+    }
+
+    return [
+      {
+        name: 'fieldConfigId',
+        fieldType: ENUM,
+        displayName: 'Field name',
+        required: true,
+        options: this.getOptions(),
+      },
+      {
+        name: 'fieldName',
+        fieldType: TEXT,
+        displayName: 'CSV field name',
+        required: true,
+      },
+      {
+        name: 'format',
+        fieldType: TEXT,
+        displayName: 'Format',
+        required: true,
+        hidden: hideFormat,
+      },
+      {
+        name: 'defaultValue',
+        fieldType: fieldConfig ? fieldConfig.fieldType : TEXT,
+        displayName: 'Default value',
+        relatedField: fieldConfig ? fieldConfig.relatedField : null,
+        relatedEntity: fieldConfig ? fieldConfig.relatedEntity : null,
+        required: fieldConfig ? fieldConfig.required : false,
+        format: fieldConfig ? fieldConfig.format : null,
+      },
+    ];
+  };
+
+  getRelatedFields = (relatedField, relatedEntity) => [
+    {
+      name: 'relatedField',
+      fieldType: RELATION,
+      displayName: 'Related field',
+      required: true,
+      relatedField,
+      relatedEntity,
+    },
+    {
+      name: 'fieldValue',
+      fieldType: TEXT,
+      displayName: 'Related field CSV name',
+      required: true,
+    },
+  ];
+
+  validate = (values) => {
+    const errors = {};
+    errors.csvFields = [];
+
+    _.forEach(values.csvFields, (csvField, key) => {
+      _.forEach(this.getFields(values.csvFields), (config) => {
```

Comment:
I don't understand this part, why you're using values.csvFields here? I think it's an array and you should use csvField instead